### PR TITLE
Implemented login and signup page.

### PR DESCRIPTION
**Pull request by Gdcr7**

### CODE REVIEW

File: lib/api/account_service.dart [hunk 1]
```diff
@@ -24,8 +25,20 @@ class AccountService
...
 
+  static Future<void> addAccount(String? id,Account account) async {
+    final accountCollectionRef = FirebaseFirestore.instance
+        .collection(collectionName)
+        .withConverter<Account>(
+            fromFirestore: Account.fromFirestore,
+            toFirestore: Account.toFirestore);
+
+
+    accountCollectionRef.doc(id).set(account);
```
[the remark this answers is not in the document]
Comment (karuniasentosa) [reply lib/api/account_service.dart:36]: Suggestion: This function could return `DocumentReference<Account>` to make it reusable inside another function. As I saw you re-fetching the same document on `ui/signup_continue_page.dart:182`

File: lib/api/account_service.dart [hunk 2]
```diff
@@ -3,8 +3,9 @@ import 'package:firebase_auth/firebase_auth.dart';
 
 import 'model/account.dart';
 
-class AccountService
-{
+class AccountService {
+  static const collectionName = 'account';
```
Comment (karuniasentosa) [inline lib/api/account_service.dart:7]: this `collectionName` is already provided in `Account` data class.

File: lib/api/account_service.dart (hunk 1 above)
[the remark this answers is not in the document]
Comment (Gdcr7) [reply lib/api/account_service.dart]: done, couldn't make sense of the suggestion though, you can tell me later if you still wanted the changes.

File: lib/ui/signup_continue_page.dart [hunk 3]
```diff
@@ -122,29 +152,111 @@ class _SignupContinuePageState extends State<SignupContinuePage> {
...
+                          .child('${args.id}.jpg');
+
+                      StorageService.putData(
+                          imageRef, image!.readAsBytesSync());
+
+                      final accountCollectionRef = FirebaseFirestore.instance
+                          .collection(Account.collectionName);
+
+                      accountCollectionRef
+                          .doc(args.id)
```
[the remark this answers is not in the document]
Comment (Gdcr7) [reply lib/ui/signup_continue_page.dart]: done

File: lib/ui/signup_page.dart [hunk 4]
```diff
@@ -88,22 +101,62 @@ class _SignupPageState extends State<SignupPage> {
...
+                    if (e.code == 'email-already-in-use') {
+                      const snackBar =
+                          SnackBar(content: Text('This email has been used'));
+                      ScaffoldMessenger.of(context).showSnackBar(snackBar);
+                    } else if (e.code == 'weak-password') {
+                      const snackBar = SnackBar(
+                        content: Text('The password provided is too weak.'),
+                      );
+                      ScaffoldMessenger.of(context).showSnackBar(snackBar);
+                    }
```
[the remark this answers is not in the document]
Comment (Gdcr7) [reply lib/ui/signup_page.dart:151]: done

File: lib/api/account_service.dart (hunk 2 above)
Comment (Gdcr7) [reply lib/api/account_service.dart]: done

**Review by karuniasentosa — APPROVED** [review]
Looks good to me!

**Pull request merged**